src/fsharp/FSharp.Core/local.fs

```diff
@@ -73,7 +73,7 @@ module internal List =
             cons
     
     let countBy (dict:Dictionary<_, int>) (keyf:'T -> 'Key) = 
-        let mutable ie = dict.GetEnumerator()
+        use mutable ie = dict.GetEnumerator()
```

Comment:
This is a bit of a performance pessimization since a dictionary's enumerator doesn't actually do anything on Dispose(); this will now emit a pointless try-finally and call to Dispose(). You could argue it's more correct to always dispose disposables, but in this case it's safe not to do it. Just my personal opinion OFC, I'm not a maintainer on this codebase.